Conversão de ficheiros .JSON e .CSV

lourencosoaress opened the pull request:
Foi adicionado os ficheiros .java relativos à conversão de ficheiro .json para .csv, .csv para .json e a classe que serve de modelo aos ficheiros .csv.
Para além disso, foram também colocadas as dependências requisitadas para o funcionamento desses ficheiros.

Commit:
Foi adicionado os ficheiros .java relativos à conversão de ficheiro .json para .csv, .csv para .json e a classe que serve de modelo aos ficheiros .csv.
Para além disso, foram também colocadas as dependências requisitadas para o funcionamento desses ficheiros.

jmmpo3 commented on src/main/java/com/iscte/engsoft/grupob/CalendarApp/util/CSVConverter.java:
```diff
@@ -0,0 +1,31 @@
+
+public class CSVConverter {
+
+	public void jsonToCSV(String filePath) {
```
Tendo em conta que esta é uma classe utilitária e não possui nenhuma propriedade de instancia este método deverá ser convertido a função (static).

jmmpo3 replied:
A classe deve seguir à risca o principio de responsabilidade única (Single Responsibility Principle) e deverá efetuar uma única coisa, neste caso deverá já receber uma String em vez de uma localização para um ficheiro.

jmmpo3 commented on src/main/java/com/iscte/engsoft/grupob/CalendarApp/util/CSV.java:
```diff
@@ -0,0 +1,276 @@
+ * 
+ * @Author Jay Sridhar
+ */
+public class CSV {
```
Eventualmente esta classe terá de provavelmente ser refactorizada uma vez que não pode ter como input um InputStream.
Tendo em conta a complexidade da classe talvez seja importante acrescentar um javadoc a explicar o contexto.

jmmpo3 commented on src/main/java/com/iscte/engsoft/grupob/CalendarApp/util/JSONConverter.java:
```diff
@@ -0,0 +1,49 @@
+
+public class JSONConverter {
+
+	public void csvToJSON(String filePath) {
```
Igual ao outro converter, deverá receber uma String.

Commit:
Foi adicionado os ficheiros .java relativos à conversão de ficheiro .json para .csv, .csv para .json e a classe que serve de modelo aos ficheiros .csv.
Para além disso, foram também colocadas as dependências requisitadas para o funcionamento desses ficheiros.
Posteriormente foram feitos os testes unitários (JUnitTest) que verificam se os inputs são válidos e se a conversão foi bem sucedida, mais uma classe abstrata que servirá de base para a estrutura do csv, permitindo fazer input do cabeçalho diretamente.

Commit:
Foi adicionado os ficheiros .java relativos à conversão de ficheiro .json para .csv, .csv para .json e a classe que serve de modelo aos ficheiros .csv.
Para além disso, foram também colocadas as dependências requisitadas para o funcionamento desses ficheiros.
Posteriormente foram feitos os testes unitários (JUnitTest) que verificam se os inputs são válidos e se a conversão foi bem sucedida, mais uma classe abstrata que servirá de base para a estrutura do csv, permitindo fazer input do cabeçalho diretamente.

Commit:
Foi adicionado os ficheiros .java relativos à conversão de ficheiro .json para .csv, .csv para .json e a classe que serve de modelo aos ficheiros .csv.
Para além disso, foram também colocadas as dependências requisitadas para o funcionamento desses ficheiros.
Posteriormente foram feitos os testes unitários (JUnitTest) que verificam se os inputs são válidos e se a conversão foi bem sucedida, mais uma classe abstrata que servirá de base para a estrutura do csv, permitindo fazer input do cabeçalho diretamente.

Commit:
Foi adicionado os ficheiros .java relativos à conversão de ficheiro .json para .csv, .csv para .json e a classe que serve de modelo aos ficheiros .csv.
Para além disso, foram também colocadas as dependências requisitadas para o funcionamento desses ficheiros.
Posteriormente foram feitos os testes unitários (JUnitTest) que verificam se os inputs são válidos e se a conversão foi bem sucedida, mais uma classe abstrata que servirá de base para a estrutura do csv, permitindo fazer input do cabeçalho diretamente.

Commit:
Foi adicionado os ficheiros .java relativos à conversão de ficheiro .json para .csv, .csv para .json e a classe que serve de modelo aos ficheiros .csv.
Para além disso, foram também colocadas as dependências requisitadas para o funcionamento desses ficheiros.
Posteriormente foram feitos os testes unitários (JUnitTest) que verificam se os inputs são válidos e se a conversão foi bem sucedida, mais uma classe abstrata que servirá de base para a estrutura do csv, permitindo fazer input do cabeçalho diretamente.

Commit:
Foi adicionado os ficheiros .java relativos à conversão de ficheiro .json para .csv, .csv para .json e a classe que serve de modelo aos ficheiros .csv.
Para além disso, foram também colocadas as dependências requisitadas para o funcionamento desses ficheiros.
Posteriormente foram feitos os testes unitários (JUnitTest) que verificam se os inputs são válidos e se a conversão foi bem sucedida, mais uma classe abstrata que servirá de base para a estrutura do csv, permitindo fazer input do cabeçalho diretamente.

Commit:
Foi adicionado os ficheiros .java relativos à conversão de ficheiro .json para .csv, .csv para .json e a classe que serve de handler aos ficheiros .csv.
Para além disso, foram também colocadas as dependências requisitadas para o funcionamento desses ficheiros.
Posteriormente foram feitos os testes unitários (JUnitTest) que verificam se os inputs são válidos e se a conversão foi bem sucedida, mais uma classe abstrata que servirá de base para a estrutura do csv, permitindo fazer input do cabeçalho diretamente.

Commit:
Foi colocada um método teste: caso a conversão foi bem sucedida (caso seja igual à conversão expectada) e se dada uma string incorreta ela lança uma exceção.

jmmpo3 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Classe utilitária para conversão de horário de CSV para JSON](https://trello.com/c/hmyPgBZ3/4-classe-utilit%C3%A1ria-para-convers%C3%A3o-de-hor%C3%A1rio-de-csv-para-json)

jmmpo3 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Classe utilitária para conversão de horário de JSON para CSV](https://trello.com/c/tBRCy7bw/5-classe-utilit%C3%A1ria-para-convers%C3%A3o-de-hor%C3%A1rio-de-json-para-csv)
